turbopack/crates/turbo-tasks-backend/src/lmdb_backing_storage.rs

```diff
@@ -60,8 +67,83 @@ pub struct LmdbBackingStorage {
 }
 
 impl LmdbBackingStorage {
-    pub fn new(path: &Path) -> Result<Self> {
-        create_dir_all(path)?;
+    pub fn new(base_path: &Path) -> Result<Self> {
+        // Database versioning. Pass `TURBO_TASKS_IGNORE_GIT_DIRTY` at compile time to ignore a
+        // dirty git repository. Pass `TURBO_TASKS_DISABLE_VERSION` at runtime to ignore a
+        // dirty git repository.
+        let git_dirty =
+            option_env!("TURBO_TASKS_IGNORE_GIT_DIRTY").is_none() && GIT_DIRTY == Some(true);
+        let disabled_versioning = env::var("TURBO_TASKS_DISABLE_VERSION").ok().is_some();
+        let version = match (disabled_versioning, git_dirty) {
+            (false, false) => GIT_COMMIT_HASH_SHORT,
+            (false, true) => {
+                println!(
+                    "WARNING: The git repository is dirty: Persistent Caching is disabled. Force \
+                     enable it with TURBO_TASKS_DISABLE_VERSION=1"
+                );
+                None
+            }
```

Comment:
I think a warning is good, but I don't think we actually need to disable persistent caching. We can just change the versioning key to be the commit hash plus a timestamp, so that persistent caching is only valid against that exact binary.

I guess that timestamp would need to be captured on the top-level next-swc binary, and not inside this library crate though?

Reply:
It's not that reliable to capture dirty flag and timestamp, since it only capture it when the build.rs is executed and that can be cached.

Reply:
> that can be cached

That's why I think this would need to happen on the top-level `next-swc` binary's `build.rs` (and get passed down to here)? Or is that not even enough?